terraform/modules/hub/hub_key_rotation.tf

```diff
@@ -0,0 +1,103 @@
+resource "aws_s3_bucket" "hkr_sam" {
+  bucket = "govukverify-hkr-${var.deployment}"
+  region = data.aws_region.region.id
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+
+  lifecycle {
+    prevent_destroy = true
+  }
+
+  tags = {
+    Environment = var.deployment
+    ManagedStackSource = "AwsSamCli"
+    ManagedBy   = "terraform"
+  }
+}
+
+data "aws_iam_policy_document" "hkr_sam_bucket_policy" {
+  statement {
+    sid    = "HKRSamBucketPolicy"
+    effect = "Allow"
+
+    principals {
+      type="AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.account.account_id}:role/accounts-deployer-role"
+      ]
+    }
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+      "s3:ListBucket"
+    ]
+    resources = [
+      "${aws_s3_bucket.hkr_sam.arn}", 
+      "${aws_s3_bucket.hkr_sam.arn}/*"
+    ]
+  }
+}
+
+resource "aws_s3_bucket_policy" "hkr_sam_policy" {
+  bucket = aws_s3_bucket.hkr_sam.id
+  policy = data.aws_iam_policy_document.hkr_sam_bucket_policy.json
+}
+
+resource "aws_ecr_repository" "hub_key_rotation" {
```

Comment:
What's this repo going to contain? Will there be a repo per account/env, or just one global image?
If the latter, it should probably go with all the other ECR repos [here](https://github.com/alphagov/verify-terraform/blob/master/environments/tools/platform/ecr.tf)

Reply:
This repo will contain the images created for each  step functions used for hub key rotation. This will be per env and multiple images.

Reply:
Is each image going to live under a different tag?